core/common/src/main/java/alluxio/MasterInquireClient.java

```diff
@@ -79,11 +80,13 @@ private MasterInquireClient(String zookeeperAddress, String electionPath, String
    * @return the address of the current leader master
    */
   public synchronized String getLeaderAddress() {
+    long startTime = System.currentTimeMillis();
     int tried = 0;
     try {
+      ZooKeeper zookeeper = mClient.getZookeeperClient().getZooKeeper();
       while (tried < mMaxTry) {
-        if (mClient.checkExists().forPath(mLeaderPath) != null) {
-          List<String> masters = mClient.getChildren().forPath(mLeaderPath);
+        if (zookeeper.exists(mLeaderPath, false) != null) {
```

Comment:
Shall we check `isConnected()` before using other methods? according to the Zookeeper doc [here](http://curator.apache.org/curator-client/index.html)